### PR TITLE
build: learn to take CustomTargetIndex as custom_target command argument

### PR DESCRIPTION
**Pull request by elmarco**
Fix ERROR: Argument <CustomTargetIndex:...>[0]> in "command" is invalid.

### CODE REVIEW

File: mesonbuild/build.py [hunk 1]
```diff
@@ -2186,6 +2186,8 @@ def flatten_command(self, cmd):
             elif isinstance(c, (BuildTarget, CustomTarget)):
                 self.dependencies.append(c)
                 final_cmd.append(c)
+            elif isinstance(c, CustomTargetIndex):
```
Comment (dcbaker) [inline mesonbuild/build.py:2189]: you also need `self.dependencies.append(c.target)` here, or you won't create a proper dependency on the target.

You'll also need to emit a `FeatureNew.single_use()` here, as this is a new feature for 0.60

**Review by dcbaker — CHANGES_REQUESTED** [review]
This also needs a release snippet, and possibly an update to the reference manual.

File: mesonbuild/build.py (hunk 1 above)
Comment (elmarco) [reply mesonbuild/build.py:2189]: done

**Comment by elmarco** [comment]
> This also needs a release snippet, and possibly an update to the reference manual.

Are you sure it's worth it? When I wrote the test I realize where my "mistake" was. I don't think people really can reach that condition easily.

**Review by dcbaker — APPROVED** [review]
I guess not? At least with the FeatureNew we have some form of documentation.

**Pull request merged**